expo/expo
# [expo-manifests] Remove reliance upon stableLegacyId/originalFullName

# Why

These fields aren't included in the EAS manifest so we must be able to use projectId instead in all places that currently rely upon them. This PR is the first step to removing those dependencies.

Closes ENG-1244.

# How

There are a few things that use these fields:
1. Push token registration. This can be converted to using projectId in cases where projectId is known (some old embedded manifests won't have projectId, and things like snacks and local development manifests also won't at this point). Support on the server for sending projectId was added in https://github.com/expo/universe/commit/ee4821defb9eedda85ea62d3ed3c1255e6f296cb.
2. Amplitude scoping. Originally I thought that we wouldn't be able to use scope key here but after further investigation and digging deeper into the code it seems that it is more used for client-side instance deduplication and storage scoping than actually sending that value to the server (It doesn't send that value to the server). Either way it should be safe to use scopeKey as scopeKey is the same as originalFullName (stableLegacyId) for all apps so far, and for apps in the future where scopeKey is random it also should be fine since amplitude doesn't need any particular string.
3. AuthSession: https://github.com/expo/expo/blob/master/packages/expo-auth-session/src/SessionUrlProvider.ts#L51. Moving us off of this dependency was omitted from this PR, but will be investigated in a follow-up. Follow-up task: https://linear.app/expo/issue/ENG-1902/enable-authsession-to-work-with-projectid
4. Expo Home environment detection: https://github.com/expo/expo/blob/master/home/utils/Environment.ts#L7. Will investigate removal in follow-up PR

Commits:
1. Android
2. iOS
3. JS (just for new push token registration using the new notifications library)

# Test Plan

For Android and iOS, it's really tough to test this code since it is no longer used (I think) in recent SDK versions? So the test for these is just building.

For JS:
1. Manually modify `expo-notifications` to `console.log` the request body for token registration.
2. `et cp expo-notifications` (to build js)
3. In custom project that uses `expo-notifications`, "link" local expo-notifications by using `file:<relative-path>` in dependency version.
4. `yarn install --force`
5. `expo publish`
6. Load published experience in Expo Go running locally on attached android device (doesn't work on simulator)
7. See that the `console.log` logs that it is using projectId for push token registration.
8. Send a push notification to the retrieved push token, see it reaches the device.
